python_modules/libraries/dagster-k8s/dagster_k8s/executor.py

```diff
@@ -301,6 +335,13 @@ def launch_step(self, step_handler_context: StepHandlerContext) -> Iterator[Dags
         deployment_name_env_var = get_deployment_id_label(container_context.run_k8s_config)
         if deployment_name_env_var:
             labels["dagster/deployment-name"] = deployment_name_env_var
+
+        owner_references = []
+        if self._enable_owner_references:
+            my_pod = self._detect_current_name_and_uid()
+            if my_pod:
+                owner_references = [OwnerReference(kind="Pod", name=my_pod[0], uid=my_pod[1])]
```

Comment:
i was reading something online about needing to set controller: true and/or blockOwnerDeletion: True in order for garbage colletion to work, but i guess your test shows that is not needed?

Reply:
That's my impression, I set `blockOwnerDeletion` initially but stripped it out and the test seemed to still work.